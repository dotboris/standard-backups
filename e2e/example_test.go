package e2e

import (
	"os/exec"
	"testing"

	"github.com/dotboris/standard-backups/internal/testutils"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/assert"
)

func TestExampleConfigDump(t *testing.T) {
	cmd := exec.Command(
		"./dist/standard-backups",
		"config", "dump",
		"--config-dir", "examples/config/etc/standard-backups",
		"--no-color",
	)
	cmd.Dir = testutils.GetRepoRoot(t)
	output, err := cmd.CombinedOutput()
	if !assert.NoError(t, err, string(output)) {
		return
	}
	snaps.MatchSnapshot(t, string(output))
}
