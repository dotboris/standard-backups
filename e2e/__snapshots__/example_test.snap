
[TestExampleConfigDump - 1]
&config.Config{
  Backends: []config.BackendManifestV1{
    config.BackendManifestV1{
      Version:         1,
      Name:            "restic",
      Description:     "",
      Bin:             "./dist/standard-backups-restic-backend",
      ProtocolVersion: 1,
    },
    config.BackendManifestV1{
      Version:         1,
      Name:            "rsync",
      Description:     "",
      Bin:             "./dist/standard-backups-rsync-backend",
      ProtocolVersion: 1,
    },
  },
  Recipes: []config.RecipeManifestV1{
    config.RecipeManifestV1{
      Version:     1,
      Name:        "examples",
      Description: "Backs up all the example files in this repository",
      Paths:       []string{
        "examples",
      },
      Hooks: config.HooksV1{
        Before:    (*config.HookV1)(nil),
        After:     (*config.HookV1)(nil),
        OnSuccess: (*config.HookV1)(nil),
        OnFailure: (*config.HookV1)(nil),
      },
    },
    config.RecipeManifestV1{
      Version:     1,
      Name:        "nextcloud",
      Description: "",
      Paths:       []string{
        "/path/to/nextcloud",
      },
      Hooks: config.HooksV1{
        Before:    (*config.HookV1)(nil),
        After:     (*config.HookV1)(nil),
        OnSuccess: (*config.HookV1)(nil),
        OnFailure: (*config.HookV1)(nil),
      },
    },
    config.RecipeManifestV1{
      Version:     1,
      Name:        "paperless",
      Description: "",
      Paths:       []string{
        "/path/to/paperless",
      },
      Hooks: config.HooksV1{
        Before:    (*config.HookV1)(nil),
        After:     (*config.HookV1)(nil),
        OnSuccess: (*config.HookV1)(nil),
        OnFailure: (*config.HookV1)(nil),
      },
    },
  },
  MainConfig: config.MainConfig{
    Version:  1,
    Backends: map[string]config.BackendConfigV1{
      "restic": config.BackendConfigV1{
        Enable: true,
      },
      "rsync": config.BackendConfigV1{
        Enable: true,
      },
    },
    Destinations: map[string]config.DestinationConfigV1{
      "local": config.DestinationConfigV1{
        Backend: "rsync",
        Options: map[string]interface {}{
          "destination-dir": "./dist/backups/local",
        },
      },
      "local-restic": config.DestinationConfigV1{
        Backend: "restic",
        Options: map[string]interface {}{
          "env": map[string]interface {}{
            "RESTIC_PASSWORD": "supersecret",
          },
          "repo": "./dist/backups/restic-local",
        },
      },
      "s3": config.DestinationConfigV1{
        Backend: "restic",
        Options: map[string]interface {}{},
      },
    },
    Jobs: map[string]config.JobConfigV1{
      "nextcloud": config.JobConfigV1{
        Recipe:   "nextcloud",
        BackupTo: []string{
          "local",
          "s3",
        },
      },
      "paperless": config.JobConfigV1{
        Recipe:   "paperless",
        BackupTo: []string{
          "s3",
        },
      },
      "test": config.JobConfigV1{
        Recipe:   "examples",
        BackupTo: []string{
          "local",
        },
      },
      "test-restic": config.JobConfigV1{
        Recipe:   "examples",
        BackupTo: []string{
          "local-restic",
        },
      },
    },
  },
}

---
